### ISSUE TITLE
Use spigot buildtools

### ISSUE THREAD

**Issue by brantje** [itzg/dockerfiles#66]
As per version 1.8 of spigot, using the buildtools is now recommend.
Here is a short how buildtools works: https://www.spigotmc.org/threads/buildtools-updates-information.42865/

And a small docker script is here: 
https://github.com/ruippeixotog/docker-spigot/blob/master/Dockerfile

Is it possible that you update your container to make use of the buildtools?

Thanks for your container, really enjoying it!

**Comment by brantje** [comment]
Working on it, currently getting permission errors for /root

**Comment by itzg** [comment]
Cool and thanks for taking a stab at it. What kind of errors are you seeing exactly? So far hadn't made use of /root...so perhaps the build needs to happen after the sudo to 'Minecraft' user. 

Now we will need to keep an eye on the build time slowing down the startup of the container...a vaguely remember that was a barrier before.

**Comment by brantje** [comment]
Where do you switch to use minecraft user?

**Comment by itzg** [comment]
It happens at the end of 'start.sh', so make sure you're putting changes into 'start-minecraft.sh'

**Comment by brantje** [comment]
Thanbks, i go fiddle with it, this is my first container i'm going to edit ^_^.

**Comment by itzg** [comment]
Enjoy. It's basically a disposable shell environment, so if you mess up just run another container :)

**Comment by brantje** [comment]
Currently it always rebuilds spigot on boot, maybe add an method to check if spigot_server.jar exists, if not start the building?

**Comment by brantje** [comment]
It works! Can you check my spigotBuildTools branch?
If everything looks ok i'll make a PR

**Comment by brantje** [comment]
@itzg i've added some ENV variables but they are not working?
Trying to figure out what i'm foing wrong.
My docker run command:
````
docker run -i -p 25565:25565 -e ENABLE_RCON=true -e RCON_PASSWORD=1234 -e MAX_PLAYERS=50 -e ENABLE_QUERY=true -e MAX_WORLD_SIZE=10000 -e OPS=brantje -e EULA=true -e TYPE=BUKKIT -v /data/docker-containers/mc-server:/data
````
And the code i'm using is [here](https://github.com/brantje/dockerfiles/commit/d2c3865be388df572c412811fad781e6e98fa762)

**Comment by itzg** [comment]
That's strange that sed wouldn't process the new ones correctly. In any case, I'll be able to take a deeper look at all your changes later today.

**Comment by brantje** [comment]
Enable query / Enable rcon works.
Howver setting a rcon_password doens't
However it is in the variables:
````
Creating server.properties
Setting motd
Enabling query
Enabling rcon
Setting rcon password to 1234
Setting max players
Setting max world size
````
Thanks for looking at my code

**Comment by brantje** [comment]
All of my ENV variables are working!

**Comment by Farigh** [comment]
Hi,
I've been messing with the container too and dispite using forge, I noticed BUKKIT is converted to SPIGOT wich might not be what the user wan't. It would be better to display a warning or error message telling it's not supported (and eventually fallback to SPIGOT anyway)

**Comment by itzg** [comment]
Good idea @Farigh about displaying a warning at least. @sshipway I remembered we talked about some of this in https://github.com/itzg/dockerfiles/pull/52 ...the current logic will always "upgrade" to Spigot, right?

**Comment by sshipway** [comment]
The thing about Bukkit is that the project has been 451'd by the powers that be, and it is no longer distributed or in active development.  Since Spigot is (supposed to be) fully compatible with Bukkit modules, converting Bukkit to Spigot makes sense.  Also, there is no reliable way to download a Bukkit JAR file at the moment whereas Spigot can be downloaded and even distributed along with the container.

**Comment by itzg** [comment]
@sshipway , I haven't dug in class by class, but at least at the top level are very slight differences in zip contents/sizes, so I'm still confused by this whole Bukkit / Spigot thing :(

![image](https://cloud.githubusercontent.com/assets/988985/14583865/042a4268-03f5-11e6-897e-e73924eb3222.png)

![image](https://cloud.githubusercontent.com/assets/988985/14583875/3180447e-03f5-11e6-9f6f-48dfdb7d6814.png)

**Comment by itzg** [comment]
Fixed by #67
